src/jailer/src/env.rs

```diff
@@ -128,9 +121,44 @@ impl Env {
 
         let daemonize = arguments.value_as_bool("daemonize").unwrap_or(false);
 
+        // Optional arguments.
+
+        let mut cgroups = Vec::new();
+
+        // if `--node` is used the corresponding cgroups will be created
+        if let Some(numa_node_str) = arguments.value_as_string("node") {
+            let numa_node = numa_node_str
+                .parse::<u32>()
+                .map_err(|_| Error::NumaNode(numa_node_str))?;
+
+            if let Ok(mut numa_cgroups) =
+                cgroup::cgroups_from_numa_node(numa_node, &id, &exec_file_name)
+            {
+                cgroups.append(&mut numa_cgroups);
+            }
+        }
+
+        // cgroup format: <cgroup_controller>.<cgroup_property>=<value>,...
+        if let Some(cgroups_args) = arguments.value_as_vector("cgroup") {
```

Comment:
Yes, for that case it will iterate over all the cgroups and writes the values (so the last ones will persist). I think that checking if a cgroup is duplicated will add unnecessary overhead for users how use this feature being conscious of what they are doing and adding just the necessary cgroups. Also AFAIK `--node` will be deprecated so this will be temporal.